src/Analyzer/Passes/ShardNumColumnToFunctionPass.cpp

```diff
@@ -49,10 +50,20 @@ class ShardNumColumnToFunctionVisitor : public InDepthQueryTreeVisitorWithContex
         if (!storage->isVirtualColumn(column.name, storage_snapshot->metadata))
             return;
 
-        auto function_node = std::make_shared<FunctionNode>("shardNum");
-        auto function = FunctionFactory::instance().get(function_node->getFunctionName(), getContext());
-        function_node->resolveAsFunction(function->build(function_node->getArgumentColumns()));
-        node = std::move(function_node);
+        const auto & column_type = column_node->getColumnType();
```

Comment:
We need to make a copy because this is a reference to `node.column` which is destroyed on line 64.
```suggestion
        const auto column_type = column_node->getColumnType();
```